include/dlaf/solver/backtransformation/mc.h

```diff
@@ -0,0 +1,284 @@
+                      Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  constexpr auto Left = blas::Side::Left;
+  constexpr auto Right = blas::Side::Right;
+  constexpr auto Upper = blas::Uplo::Upper;
+  constexpr auto Lower = blas::Uplo::Lower;
+  constexpr auto NoTrans = blas::Op::NoTrans;
+  constexpr auto ConjTrans = blas::Op::ConjTrans;
+  constexpr auto NonUnit = blas::Diag::NonUnit;
+
+  using hpx::util::unwrapping;
+
+  using hpx::execution::parallel_executor;
+  using hpx::resource::get_thread_pool;
+  using hpx::threads::thread_priority;
+
+  parallel_executor executor_hp(&get_thread_pool("default"), thread_priority::high);
+  parallel_executor executor_normal(&get_thread_pool("default"), thread_priority::default_);
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_c.blockSize().rows();
+  const SizeType nb = mat_c.blockSize().cols();
+  const SizeType ms = mat_c.size().rows();
+  const SizeType ns = mat_c.size().cols();
+
+  // Matrix T
+  comm::CommunicatorGrid comm_grid(MPI_COMM_WORLD, 1, 1, common::Ordering::ColumnMajor);
+  common::Pipeline<comm::CommunicatorGrid> serial_comm(comm_grid);
+
+  int tottaus;
+  if (ms < mb || ms == 0 || ns == 0)
+    tottaus = 0;
+  else
+    tottaus = (ms / mb - 1) * mb + ms % mb;
+
+  if (tottaus == 0)
+    return;
+
+  LocalElementSize sizeT(tottaus, tottaus);
+  TileElementSize blockSizeT(mb, mb);
+  Matrix<T, Device::CPU> mat_t(sizeT, blockSizeT);
+
+  Matrix<T, Device::CPU> mat_vv({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w({mat_v.size().rows(), mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w2({mb, mat_c.size().cols()}, mat_c.blockSize());
+
+  SizeType last_mb;
+
+  if (mat_v.blockSize().cols() == 1) {
+    last_mb = 1;
+  }
+  else {
+    if (mat_v.size().cols() % mat_v.blockSize().cols() == 0)
+      last_mb = mat_v.blockSize().cols();
+    else
+      last_mb = mat_v.size().cols() % mat_v.blockSize().cols();
+  }
+
+  Matrix<T, Device::CPU> mat_vv_last({mat_v.size().rows(), last_mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w_last({mat_v.size().rows(), last_mb}, mat_v.blockSize());
+  Matrix<T, Device::CPU> mat_w2_last({last_mb, mat_c.size().cols()}, mat_c.blockSize());
+
+  const SizeType last_reflector_idx = (mat_v.size().cols() < mat_v.size().rows())
+                                          ? mat_v.nrTiles().rows() - 2
+                                          : mat_v.nrTiles().cols() - 2;
+
+  for (SizeType k = last_reflector_idx; k > -1; --k) {

```

Comment:
Can you comment on this? In particular
- the naming of the variable
-  and why you have to do `- 2`

Moreover, minor comment (more personal style than anything else): I would do `for (... ; k >= 0; ...)`

Reply:
About the naming, in contrast with what I understood initially before looking thoroughly the code, it seems that this is not exactly what I was saying in https://github.com/eth-cscs/DLA-Future/pull/339#discussion_r602338325
> The name of the variable reflectors suggests (IMHO) that it stores the total number of reflectors, while from the computation and the for loop I would say that you are computing the last reflector index.

Reply:
For what concerns the naming, I compute the number of tiles that contain reflectors. What do you think? Any suggestion for improving the name?

The `-2` is used to take into account the fact that, for the specific layout of `V` we are now considering, the last column of tile will always be empty (full of zero values). We will have to generalize it when the layout will change (and the reduction to band will be generalized as well). I've added a comment about this.